fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt

```diff
@@ -12,29 +12,39 @@ import androidx.compose.foundation.layout.padding
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.rememberNestedScrollInteropConnection
 import androidx.compose.ui.unit.dp
 import com.google.accompanist.pager.ExperimentalPagerApi
 import com.google.accompanist.pager.HorizontalPager
 import com.google.accompanist.pager.rememberPagerState
 import kotlinx.coroutines.launch
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
 import mozilla.components.lib.state.ext.observeAsComposableState
 import org.mozilla.fenix.compose.Divider
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
 import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
 
 /**
  * Top-level UI for displaying the Tabs Tray feature.
  *
  * @param tabsTrayStore [TabsTrayStore] used to listen for changes to [TabsTrayState].
+ * @param displayTabsInGrid Whether the normal and private tabs should be displayed in a grid.
  */
-@OptIn(ExperimentalPagerApi::class)
+@OptIn(ExperimentalPagerApi::class, ExperimentalComposeUiApi::class)
```

Comment:
We'll have a bunch of these soon and the callers would need to opt in as well, I think it might be good to add them in the build.gradle file as an opt in.

Reply:
I'm not 100% certain that the callers need to opt-in (my understanding is it's just the Composable functions that use them), but how does opting-in via a `build.gradle` file work? Is it essentially a global opt-in?

Reply:
You can add the opt in compiler flag [here](https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/build.gradle#L139). It has to be fully qualified name. 

Something like:
```
"-opt-in=com.google.accompanist.pager.ExperimentalPagerApi",
"-opt-in=androidx.compose.foundation.ExperimentalFoundationApi",
```

Reply:
Gotcha gotcha. Should we file a follow-up ticket then? I can add it to the big meta ticket and the kanban board so anyone can pick it up and work on it asap.

Reply:
I'd recommend updating in this PR itself.

Reply:
My thoughts are adding the compiler flags + doing any of the refactoring work to remove all of the `@OptIn` flags is kind of out of scope for this particular ticket/PR and detract from the current changelog.

Reply:
IMO adding a compiler flag to globally opt-in to an experimental API would do a good job of reducing explicit information in the code itself. Without personally knowing the full list of `ExperimentalComposeUIApi`s for example, it could lead to a situation where code is added without the gut check of whether we _should_ be added API calls that aren't finalized. 

That said, if we are organizationally prepared to accept that we will be ingesting experimental APIs throughout the codebase, I still agree that it should be done as a follow-up ticket so that it can be more easily tracked. I also think a separate follow-up ticket should be filed suggesting periodic investigation as to whether it's still needed and a comment linking to the second follow-up should be included in the patch for the first. 

I do see that we have at least one global opt-in for experimental coroutines API, but that to me is a better usecase than Compose opt-ins because there seems to be more churn with Compose spread across many different packages.

Reply:
I also think that if the app were fully modularized and we could opt-in in a build script that was package specific there would be less risk, like what is done in A-C. Turning on an opt-in at the build level for the entire app just feels like a larger decision, to me.

Reply:
Fair point! as discussed on slack, there are other references than this file, let's do it as a separate ticket 👍

Reply:
@MatthewTighe i hear you, let's discuss on slack to not overtake this PR with this discussion. Maybe even a good topic of conversation for android engg meeting?